internal/source/git.go

```diff
@@ -182,13 +183,27 @@ func getAuthOpts(ctx context.Context, c client.Client, repo *sourcev1.GitReposit
 		return nil, fmt.Errorf("failed to configure authentication options: %w", err)
 	}
 
+	var authOpts []auth.Option
+
+	if srcOpts.tokenCache != nil {
+		involvedObject := cache.InvolvedObject{
+			Kind:      imagev1.ImageUpdateAutomationKind,
+			Name:      srcOpts.objName,
+			Namespace: srcOpts.objNamespace,
+			Operation: cache.OperationReconcile,
+		}
+		authOpts = append(authOpts, auth.WithCache(*srcOpts.tokenCache, involvedObject))
+	}
+
+	if proxyURL != nil {
+		authOpts = append(authOpts, auth.WithProxyURL(*proxyURL))
+	}
```

Comment:
This is where we should be passing proxy URL to `fluxcd/pkg/auth`, it's a provider-agnistic option (like token cache and service account). I filed this PR for removing the duplicate from `git/gogit`: https://github.com/fluxcd/pkg/pull/918